server/graphql/invoice_line/src/mutations/common_insert.rs

```diff
@@ -0,0 +1,153 @@
+use async_graphql::*;
+use graphql_core::simple_generic_errors::CannotEditInvoice;
+use graphql_core::simple_generic_errors::ForeignKey;
+use graphql_core::simple_generic_errors::ForeignKeyError;
+use graphql_core::standard_graphql_error::StandardGraphqlError;
+use graphql_types::types::InvoiceLineNode;
+use repository::InvoiceLine;
+use service::invoice_line::common_insert_line::InsertInvoiceLine as ServiceInput;
+use service::invoice_line::common_insert_line::InsertInvoiceLineError as ServiceError;
+
+use super::outbound_shipment_line::line::LocationIsOnHold;
+use super::outbound_shipment_line::line::LocationNotFound;
+use super::outbound_shipment_line::line::NotEnoughStockForReduction;
+use super::outbound_shipment_line::line::StockLineAlreadyExistsInInvoice;
+use super::outbound_shipment_line::line::StockLineIsOnHold;
+
+#[derive(InputObject)]
+#[graphql(name = "InsertInvoiceLineInput")]
+pub struct InsertInvoiceLineInput {
```

Comment:
From the name, it sounds like it would work for inbound shipments as well... 

I am tending to have two frontend versions, one for outbound and one for prescriptions. For example, if they diverted, the frontend don't needed to be refactored, just the service code...

Otherwise, is there a better name that makes it clearer that its for outbound? just keep it InsertOutbounInvoiceLineInput?

Reply:
[2ad41c9](https://github.com/openmsupply/open-msupply/pull/1970/commits/2ad41c9a2881d99cc1931c24cfae207dee6ff627)